haystack/nodes/base.py

```diff
@@ -191,7 +191,36 @@ def run_batch(
         params: Optional[dict] = None,
         debug: Optional[bool] = None,
     ):
-        pass
+        multi_query = isinstance(queries, list) and len(queries) > 0 and isinstance(queries[0], str)
+        single_query = isinstance(queries, str)
+        multi_docs_list = isinstance(documents, list) and len(documents) > 0 and isinstance(documents[0], list)
+        single_docs_list = isinstance(documents, list) and len(documents) > 0 and isinstance(documents[0], Document)
```

Comment:
The typing is quite clear so I'd simplify this a bit:

```suggestion
        single_query = isinstance(queries, str)
        single_docs_list = isinstance(documents, list) and len(documents) > 0 and isinstance(documents[0], Document)
```

That also makes some `elif` below become simply `else`. I'll note them with a suggestion.